File: baekjoon/lecture/basic_algorithm/questions/200_data_structure_1/01_q10828.py
from sys import stdin

stdin =  open("/mnt/repository/coding-test/baekjoon/lecture/basic_algorithm/questions/input.txt", "r");
next(stdin)

stack = []
for line in stdin:
    command = line.split()

    if command[0] == 'push':
        stack.append(command[1])

    elif command[0] == 'pop':
        if stack:
            print(stack.pop())
        else:
            print(-1)
    
    elif command[0] == 'size':
        print(len(stack))
    
    elif command[0] == 'empty':
        if stack:
            print(0)
        else:
            print(1)
    
    elif command[0] == 'top':
        if stack:
            print(stack[-1])
        else:
            print(-1)

File: baekjoon/lecture/basic_algorithm/questions/200_data_structure_1/01_q10828.js
const fs = require('fs');
const input = fs.readFileSync('../input.txt').toString().trim().split('\n');

input.shift();
const commands = input.map((x) => x.split(/\s+/));

class Node {
  constructor(data) {
    this.data = data;
    this.next = null;
  }
}

class Stack {
  constructor() {
    this.top = null;
    this.stackSize = 0;
  }
  push(data) {
    let node = new Node(data);
    node.next = this.top;
    this.top = node;
    this.stackSize++;
  }
  pop() {
    if (!this.top) return -1;
    const popped = this.top.data;
    this.top = this.top.next;
    this.stackSize--;
    return popped;
  }
  size() {
    return this.stackSize;
  }
  empty() {
    return this.stackSize ? 0 : 1;
  }
  peak() {
    if (!this.top) return -1;
    return this.top.data;
  }
}

const commandProcessing = (commandArr) => {
  result = '';
  let stack = new Stack();
  commandArr.forEach((command) => {
    switch (command[0]) {
      case 'push':
        stack.push(command[1]);
        break;
      case 'pop':
        result += stack.pop() + '\n';
        break;
      case 'size':
        result += stack.size() + '\n';
        break;
      case 'empty':
        result += stack.empty() + '\n';
        break;
      case 'top':
        result += stack.peak() + '\n';
        break;
      default:
        break;
    }
  });

  console.log(result.trim());
  return 0;
};

commandProcessing(commands);
